packages/core/src/oidc-utils.ts

```diff
@@ -0,0 +1,94 @@
+import * as actions_http_client from '@actions/http-client'
+import {IRequestOptions} from '@actions/http-client/interfaces'
+import {HttpClient} from '@actions/http-client'
+import {BearerCredentialHandler} from '@actions/http-client/auth'
+import {debug, setSecret} from './core'
+
+interface IOidcClient {
+
+  createHttpClient(): actions_http_client.HttpClient
+
+  getApiVersion(): string
+
+  getRuntimeToken(): string
+
+  getIDTokenUrl(): string
+
+  postCall(httpclient: actions_http_client.HttpClient, id_token_url: string, audience: string): Promise<string>
+
+  getIDToken(audience: string): Promise<string>
+}
+
+export class OidcClient implements IOidcClient {
```

Comment:
since we don't have any member variables, does it make sense to make all the methods static then we wouldn't need to do `new OidcClient()`

Reply:
I like interfaces, but if we don't use it anywhere else is the `IOidcClient` interface necessary?